addons/udes_stock/controllers/stock_location.py

```diff
@@ -138,6 +138,19 @@ def is_compatible_package_to_location(self, package_name,
                                  location_barcode)
 
         if not package_name:
-            raise ValidationError(_('You need to provide a package name.'))
-
-        return location.is_compatible_package(package_name)
+            raise ValidationError(_("You need to provide a package name."))
+
+        res = location.is_compatible_package(package_name)
+
+        Package = request.env["stock.quant.package"]
+        package = Package.get_package(package_name, no_results=True)
+
+        if package:
+            pack_location = package.current_picking_move_line_ids.filtered(
+                lambda sml: sml.qty_done > 0
+            ).mapped("location_dest_id")
+            if pack_location and (len(pack_location) > 1 or location != pack_location):
+                raise UserError(
+                    _("You should not put the contents of a package in different locations.")
+                )
```

Comment:
I'm not sure that this achieves the objective of the story. We want to ensure that damaged stock is not scanned onto a pallet of good stock and vice versa i.e. the issue doesn't arise because we're putting a package into more than one location.

Reply:
Either way, I think this logic might exist better in `is_compatible_package`

Reply:
Check that `current_picking_move_line_ids` returns the correct move lines i.e. that it filters on the `picking_id` as I don't believe that it's in the context. We may be better off using `get_move_lines` on packages with an extra domain to filter out done move lines.